File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0;
        int right = numbers.size()-1;

        while(left < right && (left >= 0 && right < numbers.size())){
            // cout << left << "  " << right << endl;
            if(target == numbers[left]+numbers[right]){
                return {left+1, right+1};
            }else if(target < numbers[left]+numbers[right]){
                right--;
            }else if(target > numbers[left]+numbers[right]){
                left++;
            }
        }
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    List<Integer> ans = new ArrayList<>();
    Integer max_depth = 0;
    public List<Integer> rightSideView(TreeNode root) {
        if(root == null){return ans;}
        traverse(root, 1);

        return ans;
    }

    public void traverse(TreeNode node, Integer curr_depth){
        if(curr_depth > max_depth){
            ans.add(node.val);
            max_depth = curr_depth;
        }
        if(node.right != null){
            traverse(node.right, curr_depth+1);
        }
        if(node.left != null){
            traverse(node.left, curr_depth+1);
        }
    }
}
